### ISSUE TITLE
mpirun: failure to launch when using SLURM PRTE PLM

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#10153]
This issue is being filed as an Open MPI issue as it is more likely to be a google hit than an issue in prte repo.
A subsequent issue will be filed against prte.

This issue impacts master.  It was introduced by commit 953c23e35f0e395c441ee35a8e27243d7fc08d94.
To reproduce this problem find a system that is running slurm (I doubt it matters what version).
Allocate two nodes.
Build OMPI master.  
run <path_to_install_path>/bin/prte.
When I do this I get
```
howardp@crusher003:~/ompi> (master)prte
--------------------------------------------------------------------------
PRTE has lost communication with a remote daemon.

  HNP daemon   : [prte-crusher003-55533@0,0] on node crusher003
  Remote daemon: [prte-crusher003-55533@0,1] on node crusher004

This is usually due to either a failure of the TCP network
connection to the node, or possibly an internal failure of
the daemon itself. We cannot recover from this failure, and
therefore will terminate the job.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
An PRTE daemon has unexpectedly failed after launch and before
communicating back to mpirun. This could be caused by a number
of factors, including an inability to create a connection back
to mpirun due to a lack of common network interfaces and/or no
route found between them. Please check network connectivity
(including firewalls and network routing requirements).
--------------------------------------------------------------------------
```
A workaround is to
```
export PRTE_MCA_plm=ssh
```
What is happening is that a prted is segfaulting in node_regex_report.  the prte_plm structure is mostly unfilled in.  The remote_spawn field is set to NULL, hence the segfault.
```
#0  0x0000000000000000 in ?? ()
[Current thread is 1 (Thread 0x7fb0293cd740 (LWP 64647))]
Missing separate debuginfos, use: zypper install glibc-debuginfo-2.31-7.30.x86_64 liblzma5-debuginfo-5.2.3-4.3.1.x86_64 libxml2-2-debuginfo-2.9.7-3.37.1.x86_64 libz1-debuginfo-1.2.11-3.21.1.x86_64
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00000000004081e9 in node_regex_report (status=0, sender=0x97afb0, buffer=0x97b0c0, tag=64, cbdata=0x60b154 <node_regex_waiting>) at prted.c:989
#2  0x00007fb028e334b6 in prte_rml_base_process_msg (fd=-1, flags=4, cbdata=0x97aea0) at rml/rml_base_msg_handlers.c:202
#3  0x00007fb02867a1e0 in event_process_active_single_queue (base=base@entry=0x61b8a0, activeq=0x61be80, max_to_process=max_to_process@entry=2147483647, endtime=endtime@entry=0x0) at event.c:1691
#4  0x00007fb02867a9df in event_process_active (base=0x61b8a0) at event.c:1783
#5  event_base_loop (base=0x61b8a0, flags=<optimized out>) at event.c:2006
#6  0x0000000000407541 in main (argc=16, argv=0x7ffd9cf3e8e8) at prted.c:800
(gdb) up
#1  0x00000000004081e9 in node_regex_report (status=0, sender=0x97afb0, buffer=0x97b0c0, tag=64, cbdata=0x60b154 <node_regex_waiting>) at prted.c:989
989	    prte_plm.remote_spawn();
(gdb) print prte_plm
$1 = {init = 0x0, set_hnp_name = 0x0, spawn = 0x0, remote_spawn = 0x0, terminate_job = 0x0, terminate_orteds = 0x0, terminate_procs = 0x0, signal_job = 0x0, finalize = 0x0}
(gdb) print prte_plm.remote_spawn
$2 = (prte_plm_base_module_remote_spawn_fn_t) 0x0
(gdb) quit
```

**Comment by hppritcha** [comment]
@awlauria possibly of interest to 5.0.x RMs

**Comment by awlauria** [comment]
Thanks Howard.

@gpaulsen @janjust

**Comment by awlauria** [comment]
@hppritcha I don't think this broke in v5.0.x because the change that broke this only went into prrte master, right?

This should be fixed in an upcoming submodule pointer update to main.

**Comment by hppritcha** [comment]
correct, not broken in v5.0.x.  Also I didn't think the prrte refactoring that caused this problem will be happening in the prrte v2.1 branch, right?

**Comment by awlauria** [comment]
Fixed by #10212.
